### PR TITLE
docs: document TSNode:byte_length()

### PR DESCRIPTION
**Pull request by gpanders**
Also update the type annotation of TSNode:id(), which returns a string,
not an integer.

### CODE REVIEW

**Comment by clason** [comment]
Are we missing any other documentation? (Is this something we could automate from https://github.com/tree-sitter/tree-sitter/blob/v0.20.8/lib/include/tree_sitter/api.h?)

File: runtime/lua/vim/treesitter/dev.lua [hunk 1]
```diff
@@ -39,7 +39,7 @@ local TSTreeView = {}
 ---@param node TSNode Starting node to begin traversal |tsnode|
 ---@param depth integer Current recursion depth
 ---@param lang string Language of the tree currently being traversed
----@param injections table<integer,TSP.Node> Mapping of node ids to root nodes of injected language trees (see
+---@param injections table<string,table> Mapping of node ids to root nodes of injected language trees (see
```
Comment (lewis6991) [inline runtime/lua/vim/treesitter/dev.lua:42]: Is there not a better type we can use instead of table?

Comment (lucario387) [reply runtime/lua/vim/treesitter/dev.lua:42]: `{ lang: string, root: TSNode }` represents the exact content stored by injections table.

Comment (lewis6991) [reply runtime/lua/vim/treesitter/dev.lua:42]: So basically a subset of `TSP.Node`? If so we can add a class for this.

Comment (gpanders) [reply runtime/lua/vim/treesitter/dev.lua]: I added `TSP.Injection` in f2d10102b6ed825c251f3cc97305767b7b4abd7f.

**Review by lewis6991 — APPROVED** [review]

**Pull request merged**